src/optimizers/bff_optimizer.py

```diff
@@ -0,0 +1,178 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+# BFF_Optimizer: a pure Bfloat16 AdamW optimizer with optional Kahan summation
+# Features direct control over momentum, variance and auxiliary compensation
+# buffer dtypes.
+# Kahan summation is used to offset Bfloat16 precision reduction for
+# the weight updates, allowing full training in BFloat16.
+
+import torch
+from torch.optim.optimizer import Optimizer
+
+
+class BFF_Optimizer(Optimizer):
```

Comment:
maybe change the name to incorporate that the algorithm is similar to / dispatches to Adam?

Reply:
agreed - changing to BFF_AdamW